### PR TITLE
Lesson10

### PR DESCRIPTION
**Pull request by mike-scythian**

### CODE REVIEW

File: NIX_Lessons/lesson10/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  <groupId>org.example</groupId>
+  <artifactId>lesson10</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>14</source>
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/pom.xml:14]: why java 14? 😄

File: NIX_Lessons/lesson10/src/main/java/nix/lessons/Cat.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package nix.lessons;
+
+
+import java.util.Random;
+
+public class Cat extends Predator{
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/src/main/java/nix/lessons/Cat.java:6]: nit: I'd recommend using the auto-formatting shortcut in Intellij https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html

File: NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    @ParameterizedTest
+    @EnumSource(DaysOfWeek.class)
+    void shouldDoesTigidyk(DaysOfWeek day){
+
+        if(day.name().equals("SUNDAY"))
+            assertEquals(kitty.scheduleCat(day), kitty.tigidykSkill(day.ordinal()));
+    }
+
+    @Test
+    void shouldSuperPowerUse(){
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java:64]:
```suggestion
    void shouldUseSuperPower(){
```

File: NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @EnumSource(DaysOfWeek.class)
+    void shouldHunt(DaysOfWeek day){
+
+        if(day.name().equals("TUESDAY") || day.name().equals("THURSDAY") || day.name().equals("SATURDAY"))
+            assertThat(kitty.scheduleCat(day)).isBetween("Prey", "Uhrrrr");
+    }
+
+    @ParameterizedTest
+    @EnumSource(DaysOfWeek.class)
+    void shouldDoesTigidyk(DaysOfWeek day){
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java:57]:
```suggestion
    void shouldDoTigidyk(DaysOfWeek day){
```

File: NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    void shouldCatsEquals(){
+
+        Cat cat = new Cat("black", 6,"highlander");
+
+        assertTrue(kitty.equals(cat));
+        assertTrue(kitty.hashCode() == cat.hashCode());
+    }
+
+    @Test
+    void shouldCatsNotEquals(){
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java:30]:
```suggestion
    void catsShouldNotBeEqual(){
```

File: NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    private static Cat kitty;
+
+    @BeforeAll
+    static void createKitty(){
+        kitty = new Cat("black",6,"highlander");
+    }
+
+    @Test
+    void shouldCatsEquals(){
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java:21]:
```suggestion
    void catsShouldBeEqual(){
```

File: NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.jupiter.params.provider.EnumSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CatClassTests {
+
+    private static Cat kitty;
+
+    @BeforeAll
+    static void createKitty(){
```
Comment (nikmikhailov13) [inline NIX_Lessons/lesson10/src/test/java/nix/lessons/CatClassTests.java:16]: 👍

**Review by nikmikhailov13 — APPROVED** [review]
LGTM 👍 
Just minor comments and suggestions.

File: NIX_Lessons/lesson10/src/main/java/nix/lessons/Cat.java (hunk 2 above)
Comment (mike-scythian) [reply NIX_Lessons/lesson10/src/main/java/nix/lessons/Cat.java:6]: I am sorry but I don't understand what you mean in this case.  Could you explain what is wrong in this fragment or file?

**Pull request merged**